src/Nox.Types/Types/CurrencyNumber/CurrencyNumber.cs

```diff
@@ -1,12 +1,72 @@
+using System;
+using System.Globalization;
+using System.Threading;
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="CurrencyNumber"/> type and value object.
+/// Supports ETL Process's and allows to get the proper <see cref="CurrencyCode"/>
+/// </summary>    
+[Serializable]
+public sealed class CurrencyNumber : ValueObject<(uint Amount, CurrencyCode CurrencyCode), CurrencyNumber>
```

Comment:
Please confirm if we should have this as uint.
If that's the case we need a From method parsing a floating-point number to uint and a ToString parsing it back to floating-point.
If not, then we should store the values as decimal or even use our own nox type QuantityValue.

Reply:
Hi @danield-ap thanks for your suggestions on my PR. I opened a question on the interactive wiki and am I waiting for confirmation about this doubt.